nix/regenerate.sh

```diff
@@ -4,9 +4,6 @@ set -euo pipefail
 
 cd $(dirname "$0")/..
 
-# Regenerate --sha256 hashes in cabal.project
-nix build .#checkCabalProject -o check-cabal-project.sh && ./check-cabal-project.sh
```

Comment:
🤔 Don't we want/need this?

Reply:
If we mistype a hash in the cabal.project and `direnv reload` we get

```shell
direnv: loading ~/IOHK/cardano-wallet/.envrc
direnv: using flake
warning: Git tree '/Users/anviking/IOHK/cardano-wallet' is dirty
warning: ignoring the client-specified setting 'system', because it is a restricted setting and you are not a trusted user
warning: ignoring the client-specified setting 'system', because it is a restricted setting and you are not a trusted user
copying '/nix/store/lnwcc18q3mhx7saikwp1v93cib179w91-source' to the storedirenv: ([/opt/homebrew/bin/direnv export fish]) is taking a while to execute. Use CTRL-C to give up.
error: hash mismatch in fixed-output derivation '/nix/store/514wdhm14ymi53aayd85qbqfzys4bsd1-cardano-wallet-agda-f3479b5.drv':
         specified: sha256-fx5hsiOJZYuo37YIMTXUWcjqtigLNvT/3Yn/wOuZpgE=
            got:    sha256-fx5hsiOJZYuo37YIMTXUWcjqtigLNvT/3Yn/wOuZpoE=
error: 1 dependencies of derivation '/nix/store/xpg5g53xkjxx3snl3zvrgbi9l9q2ky2p-source-repository-package.drv' failed to build
error:
       … while evaluating the attribute 'shell'
         at /nix/store/wzhjzixv6w1x664vhsajrckalkhgr588-source/overlays/haskell.nix:849:13:
          848|             # Default shell
          849|             shell = shellFor rawProject.args.shell;
             |             ^
          850|

       … while evaluating the attribute 'hsPkgs.shellFor'
         at /nix/store/wzhjzixv6w1x664vhsajrckalkhgr588-source/overlays/haskell.nix:693:44:
          692|               project = addProjectAndPackageAttrs rec {
          693|                   inherit (pkg-set.config) hsPkgs;
             |                                            ^
          694|                   inherit pkg-set;

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: 1 dependencies of derivation '/nix/store/r4rw99sfibcayr3318mfhv74ijmw8y0d-cardano-wallet-plan-to-nix-pkgs.drv' failed to build
```

Reply:
> Don't we want/need this?

Automatically fetching the --sha256 hashes was a more pressing issue at a time when CHaP did not yet exist. Today, I think the cost of maintaining that automation is higher than manually adding --sha256 hashes for the few repositories that we do import in this way.

Reply:
> I think the cost of maintaining that automation is higher than manually adding

We can't just keep the `inherit (pkgs) checkCabalProject` line + this call?

Reply:
Unfortunately not — the `checkCabalProject` is actually provided by the `iohkNix` input (which adds it to `pkgs` via the overlay mechanism), which has bitrotted to the point where importing it breaks. See also https://github.com/input-output-hk/iohk-nix/issues/589 .